src/states/approved_peer/mod.rs

```diff
@@ -1087,55 +1810,29 @@ impl Elder {
         Ok(())
     }
 
-    // Send message over the network.
-    fn send_signed_message(&mut self, msg: &MessageWithBytes) -> Result<(), RoutingError> {
-        let dst = msg.message_dst();
-
-        // If the message is to a single node and we have the connection info for this node, don't
-        // go through the routing table
-        let single_target = if let DstLocation::Node(node_name) = dst {
-            self.chain.get_p2p_node(node_name)
-        } else {
-            None
-        };
+    // Signs and proves the given `RoutingMessage` and wraps it in `SignedRoutingMessage`.
+    fn to_accumulating_message(
+        &self,
+        dst: DstLocation,
+        variant: Variant,
+        node_knowledge_override: Option<u64>,
+    ) -> Result<AccumulatingMessage> {
+        let proof = self.chain.prove(&dst, node_knowledge_override);
+        let pk_set = self.chain.our_section_bls_keys().clone();
+        let secret_key = self.chain.our_section_bls_secret_key_share()?;
 
-        let (target_p2p_nodes, dg_size) = if let Some(target) = single_target {
-            (vec![target.clone()], 1)
-        } else {
-            self.get_targets(dst)?
+        let content = PlainMessage {
+            src: *self.our_prefix(),
+            dst,
+            variant,
         };
 
-        trace!(
-            "{}: Sending message {:?} via targets {:?}",
-            self,
-            msg,
-            target_p2p_nodes
-        );
-
-        let targets: Vec<_> = target_p2p_nodes
-            .into_iter()
-            .filter(|p2p_node| {
-                self.msg_filter
-                    .filter_outgoing(msg, p2p_node.public_id())
-                    .is_new()
-            })
-            .map(|node| *node.peer_addr())
-            .collect();
-
-        let cheap_bytes_clone = msg.full_bytes().clone();
-        self.send_message_to_targets(&targets, dg_size, cheap_bytes_clone);
-
-        Ok(())
-    }
-
-    /// Vote for a user-defined event.
-    pub fn vote_for_user_event(&mut self, event: Vec<u8>) {
-        self.vote_for_event(AccumulatingEvent::User(event));
+        AccumulatingMessage::new(content, secret_key, pk_set, proof)
     }
 
-    /// Returns the set of peers that are responsible for collecting signatures to verify a message;
-    /// this may contain us or only other nodes. If our signature is not required, this returns
-    /// `None`.
+    // Returns the set of peers that are responsible for collecting signatures to verify a message;
+    // this may contain us or only other nodes. If our signature is not required, this returns
+    // `None`.
```

Comment:
The function actually returns an empty vector ?

Reply:
Actually it never returns empty vec, even if we are not among the signature targets. Will fix the comment.